src/openrct2-ui/scripting/CustomMenu.cpp

```diff
@@ -33,8 +46,169 @@ namespace OpenRCT2::Scripting
+            }
+            else if (tileElement != nullptr)
+            {
+                int32_t index = 0;
+                auto el = map_get_first_element_at(mapCoords);
+                if (el != nullptr)
+                {
+                    do
+                    {
+                        if (el == tileElement)
+                        {
+                            obj.Set("tileElementIndex", index);
+                            break;
+                        }
+                        index++;
+                    } while (!(el++)->IsLastForTile());
+                }
+            }
+            auto eventArgs = obj.Take();
+
+            auto& scriptEngine = GetContext()->GetScriptEngine();
+            std::vector<DukValue> args;
+            args.push_back(eventArgs);
+            scriptEngine.ExecutePluginCall(Owner, dukHandler, args, false);
+        }
+    }
+
+    void InitialiseCustomTool(ScriptEngine& scriptEngine, const DukValue& dukValue)
+    {
+        try
+        {
+            if (dukValue.type() == DukValue::Type::OBJECT)
+            {
+                CustomTool customTool;
+                customTool.Owner = scriptEngine.GetExecInfo().GetCurrentPlugin();
+                customTool.Id = dukValue["id"].as_string();
+                customTool.Cursor = StringToCursor(dukValue["cursor"].as_string());
+                customTool.onStart = dukValue["onStart"];
+                customTool.onDown = dukValue["onDown"];
+                customTool.onMove = dukValue["onMove"];
+                customTool.onUp = dukValue["onUp"];
+                customTool.onFinish = dukValue["onFinish"];
+
+                auto toolbarWindow = window_find_by_class(WC_TOP_TOOLBAR);
+                if (toolbarWindow != nullptr)
+                {
+                    // Use a widget that does not exist on top toolbar but also make sure it isn't -1 as that
+                    // prevents abort from being called.
+                    rct_widgetindex widgetIndex = -2;
+                    tool_cancel();
+                    tool_set(toolbarWindow, widgetIndex, static_cast<TOOL_IDX>(customTool.Cursor));
+                    ActiveCustomTool = std::move(customTool);
+                    ActiveCustomTool->Start();
+                }
+            }
+        }
+        catch (const DukException&)
+        {
+            duk_error(scriptEngine.GetContext(), DUK_ERR_ERROR, "Invalid parameters.");
+        }
+    }
+
+    static constexpr const char* CursorNames[] = {
+        "arrow",         "blank",      "up_arrow",      "up_down_arrow", "hand_point", "zzz",         "diagonal_arrows",
+        "picker",        "tree_down",  "fountain_down", "statue_down",   "bench_down", "cross_hair",  "bin_down",
+        "lamppost_down", "fence_down", "flower_down",   "path_down",     "dig_down",   "water_down",  "house_down",
+        "volcano_down",  "walk_down",  "paint_down",    "entrance_down", "hand_open",  "hand_closed",
+    };
```

Comment:
This probably will fail `clang-format`, but in any case there's no point in creating a plain array, use an `std::array` or a `vector`

Reply:
This is the work of `clang-format`.